libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,486 @@
+        for (let compcve of _.get(component, `vulnerabilities.vulnerability`)) {
+          if (_.get(compcve, 'cve_id') === currcve) {
+            hascve = true;
+            location +=` ${_.get(component,`file_paths.file_path.value`)}`;
+            _.set(vuln, `paths`, location);
+          }
+        }
+      }
+      else{
+        if (_.get(component, 'vulnerabilities.vulnerability.cve_id') === currcve) {
+          hascve = true;
+          location +=` ${_.get(component,`file_paths.file_path.value`)}`;
+          _.set(vuln, `paths`, location);
+        }
+
+      }
+      if (hascve === true) {
+        const proxy = _.cloneDeep(component)
+        components.push( _.omit(proxy, `vulnerabilities`));
+      }
+    }
+    _.set(vuln, `components`, components);
+  }
+  return vulnarr
+}
+
+
+function parseXml(xml: string) {
+  const options = {
+    attributeNamePrefix: '',
+    textNodeName: 'text',
+    ignoreAttributes: false
+  };
+
+  const parsedXML = parser.parse(xml, options);
+  if (_.has(parsedXML, 'summaryreport')) {
+    throw new Error('Current mapper does not accept summary reports');
+  }
+  const arrayedControls = _.get(parsedXML, 'detailedreport.severity').map(
+    (control: {category: unknown; level: string}) => {
+      if (Array.isArray(control.category)) {
+        return {level: control.level, category: control.category};
+      } else if (!control.category) {
+        return {level: control.level};
+      } else {
+        return {level: control.level, category: [control.category]};
+      }
+    }
+  );
+  _.set(parsedXML, 'detailedreport.severity', arrayedControls);
+
+  return parsedXML;
+}
+
+function controlMappingCwe(severity: number): MappedTransform<ExecJSON.Control & ILookupPath, ILookupPath> {
+  return {
+    id: {path: 'categoryid'},
+    title: {path: 'categoryname'},
+    desc: {transformer: formatDesc},
+    descriptions: [
+      {
+        data: {transformer: formatRecommendations},
+        label: 'fix'
+      }
+    ],
+    impact: impactMapping(severity),
+    refs: [],
+    tags: {
+      cweid: {transformer: formatCweData},
+      cweDescription: {transformer: formatCweDesc},
+      nist: {transformer: nistTag}
+    },
+    source_location: {
+      ref: {
+        path: 'cwe',
+        transformer: formatSourceLocation
+      },
+    },
+    results: [
+      {
+        path: 'cwe',
+        pathTransform: getFlaws,
+        status: ExecJSON.ControlResultStatus.Failed,
+        code_desc: {transformer: formatCodeDesc},
+        start_time: {path: '$.detailedreport.first_build_submitted_date'},
+        message: {path: 'exploitability_adjustments.exploitability_adjustment.note'}
+      }
+    ]
+  };
+}
+
+export class VeracodeMapper extends BaseConverter {
+  mappings: MappedTransform<ExecJSON.Execution & {passthrough: unknown}, ILookupPath> = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion,
+      target_id: ''
+    },
+    passthrough: {
```

Comment:
this didn't need to be specified as such.  just call it `data_remainder` and then make the contents the full data but with an omit around it to get rid of the specific stuff that you used.

Reply:
changed passthrough to dataremainder, added formatPassthorughData